subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaIntegTest.groovy

```diff
@@ -135,6 +135,62 @@ class MavenPublishJavaIntegTest extends AbstractMavenPublishIntegTest {
             "publishTest-1.9.jar", "spring-core-2.5.6.jar"]
     }
 
+    def "can publish java-library with strict dependencies"() {
+        given:
+        createBuildScripts("""
+
+            ${jcenterRepository()}
+
+            dependencies {
+                api "org.springframework:spring-core:2.5.6"
+                implementation("commons-collections:commons-collections") {
+                    version { strictly '3.2.2' }
+                }
+            }
+
+            publishing {
+                publications {
+                    maven(MavenPublication) {
+                        from components.java
+                    }
+                }
+            }
+""")
+
+        when:
+        run "publish"
+
+        then:
+        javaLibrary.assertPublished()
+
+        javaLibrary.parsedPom.scopes.keySet() == ["compile", "runtime"] as Set
+        javaLibrary.parsedPom.scopes.compile.assertDependsOn("org.springframework:spring-core:2.5.6")
+        javaLibrary.parsedPom.scopes.runtime.assertDependsOn("commons-collections:commons-collections:3.2.2")
+
+        def apiVariant = javaLibrary.parsedModuleMetadata.variant('api')
+        apiVariant.dependencies.size() == 1
+        apiVariant.dependencies[0].group == 'org.springframework'
+        apiVariant.dependencies[0].module == 'spring-core'
+        apiVariant.dependencies[0].version == '2.5.6'
+        apiVariant.dependencies[0].rejectsVersion == []
+
+        def runtimeVariant = javaLibrary.parsedModuleMetadata.variant('runtime')
+        runtimeVariant.dependencies.size() == 2
+        runtimeVariant.dependencies[0].group == 'commons-collections'
+        runtimeVariant.dependencies[0].module == 'commons-collections'
+        runtimeVariant.dependencies[0].version == '3.2.2'
+        runtimeVariant.dependencies[0].rejectsVersion == [']3.2.2,)']
+        runtimeVariant.dependencies[1].group == 'org.springframework'
+        runtimeVariant.dependencies[1].module == 'spring-core'
+        runtimeVariant.dependencies[1].version == '2.5.6'
+        runtimeVariant.dependencies[1].rejectsVersion == []
```

Comment:
It would be great to have a fixture for verifying dependencies, that would be able to apply to any publication and validate POM/Ivy/.module metadata. I'll try to add something for the "publish excludes" story, if you don't do it.

```
javaLibrary.hasApiDependency('commons-collections:commons-collections:3.2.2') {
    rejectsVersion [']3.2.2,)']
    excludes ['org:foo', '*:bar']
}
```

The excludes would be validated with both `POM` and `.module`, the `rejectsVersion` would only check the `.module`.

Reply:
Ok I'll see if I have time tomorrow.